File: leetcode_704.py
# Binary Search
class Solution:
    def search(self, nums: [int], target: int) -> int:
        low, high = 0, len(nums)-1
        while low <= high:
            mid = (low+high)//2
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                high = mid - 1
            else:
                low = mid + 1
        return -1


s = Solution()
print(s.search(nums=[-1, 0, 3, 5, 9, 12], target=9))
print(s.search(nums=[-1, 0, 3, 5, 9, 12], target=2))

File: leetcode_704.js
console.log("In console - leetcode_704.js")

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
var low = 0, high = nums.length-1;
while (low <= high){
    mid = (low+high)//2
        if (nums[mid] == target)
            return mid
        else if (nums[mid] > target)
            high = mid - 1
        else
            low = mid + 1
		}
return -1;
};

console.log(search(nums = [-1,0,3,5,9,12], target = 9));
console.log(search(nums = [-1,0,3,5,8,12], target = 9));
